pipeline/src/pipeline.rs

```diff
@@ -1095,8 +1095,8 @@ impl<T: FieldElement> Pipeline<T> {
 
         let backend = factory
             .create(
-                pil.borrow(),
-                &fixed_cols[..],
+                pil.clone(),
+                fixed_cols.clone(),
```

Comment:
All this cloning will probably be quite slow in practice probably, no?

Reply:
No, because it's cloning `Arc`s!

Reply:
ah true